microhapulator/filter.py

```diff
@@ -0,0 +1,118 @@
+# -------------------------------------------------------------------------------------------------
+# Copyright (c) 2024, DHS.
+#
+# This file is part of MicroHapulator (https://github.com/bioforensics/microhapulator) and is
+# licensed under the BSD license: see LICENSE.txt.
+#
+# This software was prepared for the Department of Homeland Security (DHS) by the Battelle National
+# Biodefense Institute, LLC (BNBI) as part of contract HSHQDC-15-C-00064 to manage and operate the
+# National Biodefense Analysis and Countermeasures Center (NBACC), a Federally Funded Research and
+# Development Center.
+# -------------------------------------------------------------------------------------------------
+from Bio import SeqIO
+from microhapulator import open as mhopen
+import gzip
+
+
+class PairedReadFilter:
+    def __init__(self, r1_in, r2_in, output_prefix):
+        self.r1_in = mhopen(r1_in, "r")
+        self.r2_in = mhopen(r2_in, "r")
+        self.get_output_files(output_prefix)
+        self.num_r1_failed = 0
+        self.num_r2_failed = 0
+        self.num_both_failed = 0
+        self.num_pairs_passed = 0
```

Comment:
`self.get_output_files()` is my attempt to eliminate the r1_out, r2_out appearing without being passed in as arguments to or being derived in the parent class `__init__` .  I considered making a class for the output files, but I don't think the class would be doing anything besides just storing the file handles so a dictionary with the output files seems sufficient.

Reply:
I actually think that would be a great idea. I just submitted a PR into this branch as part of my review. See what you think of my suggestions. I also created a dedicated class for counts, and these both cleaned up the filter implementation quite a bit.